screeps-game-api/src/js_collections/js_map.rs

```diff
@@ -1,13 +1,53 @@
 //! [`JsMap`]
-use std::{collections::HashMap, marker::PhantomData};
+use std::{borrow::Borrow, collections::HashMap, marker::PhantomData};
 
-use stdweb::Object;
+use stdweb::{InstanceOf, Object, Reference, ReferenceType, Value};
+
+use {
+    traits::{FromExpectedType, IntoExpectedType, TryFrom, TryInto},
+    ConversionError,
+};
 
 pub struct JsMap<K, V> {
     inner: Object,
     phantom: PhantomData<HashMap<K, V>>,
 }
 
+impl<K, V> JsMap<K, V> {
+    pub fn len(&self) -> usize {
+        self.inner.len()
+    }
+}
+
+impl<K, V> JsMap<K, V>
+where
+    Option<V>: FromExpectedType<Value>,
+{
+    /// Gets an item, panicking if the types don't match and `check-all-casts` is enabled.
+    pub fn get<Q: ?Sized>(&self, key: &Q) -> Option<V>
+    where
+        K: Borrow<Q>,
+        for<'a> &'a Q: Into<Value>,
+    {
+        js_unwrap_ref!(@{self.inner.as_ref()}[@{key.into()}])
+    }
+
+    /// Gets an item, returning an error if the types don't match and `check-all-casts` is enabled.
+    pub fn try_get<Q: ?Sized>(&self, key: &Q) -> Option<Result<V, ConversionError>>
+    where
+        K: Borrow<Q>,
+        for<'a> &'a Q: Into<Value>,
+    {
+        match (js! {return @{self.inner.as_ref()}[@{key.into()}].into_expected_type();})
+            .into_expected_type()
+        {
+            Ok(Some(v)) => Some(Ok(v)),
```

Comment:
Oh! I didn't realize that existed now- would definitely be useful. I can add a note to switch to this once its stabilized- should be 1.33.0 according to https://github.com/rust-lang/rust/pull/57567/files?

As of stdweb 0.4.13, the project seems to be usable on stable rust, and if we can keep that it could be useful.

Reply:
The `TryFrom` trait got stabilized?